### PR TITLE
Push to Docker when running on "main"

### PR DESCRIPTION
**Pull request by nicolas-duvauchel**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -8,10 +8,13 @@ on:
 
 env:
   TEST_TAG: corefiling/pdf2html:test
+  CI_TAG: corefiling/pdf2html:ci-${{ github.run_id }}-${{ github.run_number }}
+  TAG_TAG: corefiling/pdf2html:${{ github.ref_name }}
 
 jobs:
   docker:
     runs-on: ubuntu-latest
+    environment: "PDF2HTML Public Upload Env"
```
Comment (nicolas-duvauchel) [inline .github/workflows/docker.yml:17]: The secrets used to login into Docker are specified in that environment.
Feels like there should be a way to have secrets not tied to a specific environment but that's the way swhite set it up for now.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -8,10 +8,13 @@ on:
 
 env:
   TEST_TAG: corefiling/pdf2html:test
+  CI_TAG: corefiling/pdf2html:ci-${{ github.run_id }}-${{ github.run_number }}
```
Comment (nicolas-duvauchel) [inline .github/workflows/docker.yml:11]: Generic CI tag used when changes are pushed to 'main'

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -8,10 +8,13 @@ on:
...
+  TAG_TAG: corefiling/pdf2html:${{ github.ref_name }}
```
Comment (nicolas-duvauchel) [inline .github/workflows/docker.yml:12]: Used when a tag is pushed

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -36,3 +39,16 @@ jobs:
...
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASS }}
+      - if: github.ref_name == 'main' || github.ref_type == 'tag'
+        name: Push to Docker Hub
+        uses: docker/build-push-action@v4
+        with:
+          context: ./src/Pdf2Html
+          push: true
+          tags: ${{ github.ref_type == 'tag' && env.TAG_TAG || env.CI_TAG }}
```
Comment (nicolas-duvauchel) [inline .github/workflows/docker.yml:54]: No ternary operator in Github actions 😞

File: .github/workflows/docker.yml (hunk 1 above)
Comment (ethanbishop) [reply .github/workflows/docker.yml:17]: Yep, let's leave it like this for now then.

**Review by ethanbishop — APPROVED** [review]

**Pull request merged**